### PR TITLE
Add feature to delete flags after they are more than 2 weeks old

### PR DESCRIPTION
**Pull request by jerryli0273**

### CODE REVIEW

File: capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    /** delete flags if the date property of the flag is before
+      * the previously calculated date. 
+      */
+    for (Entity entity: results.asIterable()) {
+        Date flagDate = (Date)entity.getProperty("date");
+        if (currentDateMinusTwoWeeks.compareTo(flagDate) > 0) {
+            datastore.delete(entity.getKey());
+        }
+        if (currentDateMinusTwoWeeks.compareTo(flagDate) >= 0) {
```
Comment (mk-valentine) [inline capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java]: maybe just 
else { 
  break; 
}
here?

File: capstone/src/main/webapp/flags_script.js [hunk 2]
```diff
@@ -143,16 +143,21 @@ function generateMap() {
   });
 }
  
-/* Populate the maps with flags. */
+/* Populate the maps with flags from the data in datastore. */
 async function getFlags() {
+  checkLogin(); //call function to hide login/logout buttons
+  deleteExpiredFlags(); //delete flags that are more 14 days old
```
Comment (mk-valentine) [inline capstone/src/main/webapp/flags_script.js:149]: how often does this run, how many flags would it potentially load, is it only deleting expired flags for this user? might be okay to leave as is for now, but if you were building this for production, you would want a job separate from the web app dealing with deleting the expired flags

File: capstone/src/main/webapp/login_script.js [hunk 3]
```diff
@@ -1,10 +1,14 @@
+/** Logs the user basic info on the console once the users
```
Comment (mk-valentine) [inline capstone/src/main/webapp/login_script.js:1]: nice comments!

**Review by mk-valentine — APPROVED** [review]
Looks good!

File: capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java (hunk 1 above)
Comment (jerryli0273) [reply capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java]: yeah that's definitely an error. I changed it to the method you suggested

File: capstone/src/main/webapp/flags_script.js (hunk 2 above)
Comment (jerryli0273) [reply capstone/src/main/webapp/flags_script.js:149]: That makes sense. For right now, the getFlags() method gets called once every time the website is loaded, not for every flag added. I will try to figure out how to have datastore delete entities serparately

Comment (foodhype) [reply capstone/src/main/webapp/flags_script.js:149]: You could just pull flags, check that they were created in the past two weeks, and then have a separate regular batch process for deleting expired flags.

File: capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.text.ParseException;
+
+/** Servlet responsible for deleting expired flags. */
```
Comment (foodhype) [inline capstone/src/main/java/com/google/sps/servlets/DeleteFlags.java:37]: This should probably be a batch process that runs on a regular schedule. For read_flags, you can read all nearby flags and then only keep the ones that were created in the last two weeks.

File: capstone/target/capstone-1/WEB-INF/appengine-generated/datastore-indexes-auto.xml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<!-- Indices written at Wed, 5 Aug 2020 17:57:13 UTC -->
```
Comment (foodhype) [inline capstone/target/capstone-1/WEB-INF/appengine-generated/datastore-indexes-auto.xml:1]: This file should probably be in .gitignore

File: capstone/target/maven-archiver/pom.properties [hunk 6]
```diff
@@ -1,5 +1,5 @@
 #Generated by Maven
```
Comment (foodhype) [inline capstone/target/maven-archiver/pom.properties:1]: Generated files should generally be in .gitignore

File: capstone/target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst [hunk 7]
```diff
@@ -3,3 +3,4 @@ com/google/sps/servlets/reCaptchaVerify.class
 com/google/sps/data/Flag.class
```
Comment (foodhype) [inline capstone/target/maven-status/maven-compiler-plugin/compile/default-compile/createdFiles.lst:3]: The entire target directory can probably be in .gitignore.

File: capstone/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst [hunk 8]
```diff
@@ -1,5 +1,6 @@
-/home/micyohan/Capstone/capstone/src/main/java/com/google/sps/servlets/PlacesServlet.java
```
Comment (foodhype) [inline capstone/target/maven-status/maven-compiler-plugin/compile/default-compile/inputFiles.lst:1]: The entire target directory can probably be in .gitignore.

**Pull request merged**